packages/eth/src/websocket/connection.rs

```diff
@@ -106,6 +110,15 @@ impl WsConnection {
         let contract_address = Address::from_slice(contract_address.as_ref());
         let contract = FUEL_STATE_CONTRACT::new(contract_address, Arc::new(signer));
 
+        let interval_u256 = contract.blocks_per_commit_interval().call().await?;
+
+        let commit_interval = u32::try_from(interval_u256)
+            .map_err(|e| Error::Other(e.to_string()))
+            .and_then(|value| {
+                NonZeroU32::new(value)
+                    .ok_or_else(|| Error::Other("Value cannot be zero".to_string()))
```

Comment:
You should probably give a bit more context here. Otherwise the error is just going to be "value cannot be zero" without saying which value and what the value means.

Maybe something like
```suggestion
                    .ok_or_else(|| Error::Other("l1 contract reported a commit interval of 0"))
```